cmd/pj-rehearse/main.go

```diff
@@ -161,14 +164,42 @@ func main() {
 		}
 	}
 
+	changedTemplates := make(map[string]*templateapi.Template)
 	// We can only detect changes if we managed to load both CI template versions
 	if masterConfig.Templates != nil && prConfig.Templates != nil {
-		changedTemplates := diffs.GetChangedTemplates(masterConfig.Templates, prConfig.Templates, logger)
+		changedTemplates = diffs.GetChangedTemplates(masterConfig.Templates, prConfig.Templates, logger)
 		for name := range changedTemplates {
 			logger.WithField("template-name", name).Info("Changed template")
 		}
 	}
 
+	var cmClient corev1.ConfigMapInterface
+	if !o.dryRun {
+		cmClient, err = rehearse.NewCMClient(clusterConfig, masterConfig.Prow.ProwJobNamespace, o.dryRun)
+		if err != nil {
+			logger.WithError(err).Error("could not create a configMap client")
+			gracefulExit(o.noFail, misconfigurationOutput)
+		}
+
+		cmController := config.NewTemplateCMController(cmClient, prNumber, logger, changedTemplates)
+		if errors := cmController.CreateCMTemplates(); len(errors) > 0 {
+			for _, err := range errors {
+				logger.WithError(err).Error("couldn't create template configMap")
+			}
+			gracefulExit(o.noFail, "")
```

Comment:
I think the cleanup here is a little more complicated - `CreateCMTemplates` might have created few CMs successfully before a failure occurred, so we would leave thes on the cluster.

Reply:
Good point. So we should ignore `doesn't exist` errors

Reply:
I changed the cleanup process to delete a collection of a configMaps that have the specific label.